fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4

```diff
@@ -1230,13 +1233,15 @@ grantPrivilegeStatement
     | GRANT privilegeActionList ON tableDbPrivilegeObjectNameList TO grantRevokeClause             #grantTablePrivBrief
     | GRANT privilegeActionList ON privilegeType (privilegeObjectNameList)? TO grantRevokeClause   #grantPrivWithType
     | GRANT privilegeActionList ON ALL privilegeType (IN ALL privilegeType)* (IN privilegeType identifierOrString)? TO grantRevokeClause   #grantOnAll
+    | GRANT privilegeActionList ON FUNCTION qualifiedName '(' typeList ')' TO grantRevokeClause             #grantPrivWithFunc
```

Comment:
why not reuse `privilegeType` and if we write like this, can we still support `grant xxx on all functions...`?

Reply:
refer to this doc (https://starrocks.feishu.cn/docx/FHCHdcSrLo08Dpx9LxtcplPkneg ) (function related) to check all the grant/revoke statements we need to support for function/mv object and we should also add test cases for them since it's new grammar.